### PR TITLE
templates: addition of OAuth login buttons macro

### LINKED ISSUE #26: global: check-up and clean
- [ ] Check that settings nicely integrate with accounts settings (Linked accounts)
- [ ] Add template macros so that login buttons for each service appears the login page.

### PR DESCRIPTION
**Pull request by JavierDelgadoFernandez**
* Adds a macro to generate the links to connect with the installed
  OAuth services.

* Adds a login_user.html template with the form showing the
  OAuth services to login and the fields to a regular login.
  (closes #26)

Co-Authored-By: Jiri Kuncar <jiri.kuncar@cern.ch>
Signed-off-by: Javier Delgado <javier.delgado.fernandez@cern.ch>

### CODE REVIEW

File: invenio_oauthclient/templates/invenio_oauthclient/_macros.html [hunk 1]
```diff
@@ -48,3 +48,31 @@
...
 {% endmacro %}
+
+
+{% macro oauth_services() %}
+  {% set next = request.script_root + request.path if request.endpoint != 'webaccount.logout' and request.endpoint != 'webaccount.login' else config.CFG_SITE_SECURE_URL %}
+
+  {% if 'github' in config.OAUTHCLIENT_REMOTE_APPS %}
+      <br />
+      <a href="{{url_for('invenio_oauthclient.login', remote_app='github', next=next)}}" class="btn btn-default btn-lg btn-block">
+        <i class="fa fa-github"/> Sign in with GitHub
```
Comment (jirikuncar) [inline invenio_oauthclient/templates/invenio_oauthclient/_macros.html]: please mark *all* strings in templates as translatable

**Comment by jirikuncar** [comment]
- What about iterating over keys in `OAUTHCLIENT_REMOTE_APPS`?
- IMHO using `<br/>` makes it hard to style
- I would propose to create custom css class for `ORCID` instead of custom template.
- Inspiration http://lipis.github.io/bootstrap-social/

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Rebased and updated.

**Comment by jirikuncar** [comment]
superseded by #32

**Pull request closed without merging**